stanzas/table-with-pagination/app.vue

```diff
@@ -404,161 +333,95 @@ export default defineComponent({
-          }
-          pageSlider.setPage(dragX, page);
-        }
-      },
-      up: (e) => {
-        if (state.knobDrag) {
-          state.knobX += e.pageX - state.startX;
-          state.pagination.currentPage = parseInt(state.knob.innerHTML);
-          state.knobDrag = false;
-        }
-      },
-      click: () => {
-        state.knobX =
-          (state.sliderBarWidth / (totalPages.value - 1)) *
-          (state.pagination.currentPage - 1);
-        pageSlider.setPage(state.knobX, state.pagination.currentPage);
-      },
-      setPage: (knobX, page) => {
-        state.knob.innerHTML = page;
-        state.knob.parentNode.style.transform = "translateX(" + knobX + "px)";
-        state.canvas.setAttribute("width", state.sliderBarWidth);
-        state.canvas.setAttribute("height", 50);
-        let pageButton = state.canvas.parentNode.parentNode
-          .getElementsByClassName("paginationWrapper")[0]
-          .getElementsByTagName("ul")[0];
-        if (state.canvas.getContext) {
-          const ctx = state.canvas.getContext("2d");
-          ctx.clearRect(
-            0,
-            0,
-            state.canvas.offsetWidth,
-            state.canvas.offsetHeight
+    function isSearchOn(column) {
+      switch (column.searchType) {
+        case "number":
+          return (
+            column.minValue !== column.rangeMinMax[0] ||
+            column.maxValue !== column.rangeMinMax[1]
           );
-          ctx.beginPath();
-          ctx.moveTo(knobX + state.knob.offsetWidth / 2 - 8, 50);
-          ctx.lineTo(knobX - state.knob.offsetWidth / 2 + 8, 50);
-          ctx.lineTo(
-            pageButton.offsetLeft - pageButton.parentNode.offsetLeft - 10,
-            0
+        default:
+          return (
+            state.queryByColumn.column === column.label &&
+            state.queryByColumn.query !== ""
           );
-          ctx.lineTo(
-            pageButton.offsetLeft -
-              pageButton.parentNode.offsetLeft +
-              pageButton.offsetWidth -
-              10,
-            0
-          );
-          ctx.closePath();
-          ctx.fillStyle = "#dddddd";
-          ctx.fill();
-        }
-      },
-    };
+      }
+    }
 
     async function fetchData() {
-      // const res = await fetch(params.table_data_api);
+      // const res = await fetch(params["table-data-api"]);
       // const data = await res.json();
 
       state.responseJSON = data;
-
-      const { vars, labels, order, href } = data.head;
-
-      const columns = zip(vars, labels, order, href)
-        .map(([_var, label, _order, _href]) => {
-          const values = data.body.map((row) => row[_var].value);
-          const datam = data.body[0];
-          if (datam[_var].type === "number") {
-            state.rangeInputs[_var] = { value: [0, 0], min: 0, max: 0 };
-          }
+      let columns = [];
+      if (params.columns) {
+        columns = JSON.parse(params.columns);
+      } else if (data.length !== 0) {
```

Comment:
条件によって columns が変わったり変わらなかったりするのがわかりにくいので、以下のようにしてはどうでしょうか。

```suggestion
      let columns;
      
      if (params.columns) {
        columns = JSON.parse(params.columns);
      } else if (data.length > 0) {
        // ...
      } else {
        columns = [];
      }
```

Reply:
何度もご指示いただいてしまいすみません。指摘内容理解しました。Suggested changeのように書き換えております。